packages/core/admin/admin/src/content-manager/components/DynamicTable/BulkActionsBar/index.js

```diff
@@ -38,12 +43,50 @@ const BulkActionsBar = ({
       handleToggleShowDeleteAllModal();
     }
   };
+  /**
+   * @param {number[]} selectedIds - Array of ids to publish
+   * @returns {{validIds: number[], errors: Object.<number, string>}} - Returns an object with the valid ids and the errors
+   */
+  const validateEntriesToPublish = async () => {
+    const validations = { validIds: [], errors: {} };
+    // Create the validation schema based on the contentType
+    const schema = createYupSchema(contentType, { components }, { isDraft: false });
+    // Get the selected entries
+    const entries = data.filter((entry) => {
+      return selectedEntries.includes(entry.id);
+    });
+    // Validate each entry and map the unresolved promises
+    const validationPromises = entries.map((entry) =>
+      schema.validate(entry, { abortEarly: false })
+    );
+    // Resolve all the promises in one go
+    const resolvedPromises = await Promise.allSettled(validationPromises);
```

Comment:
will this not result in MySQL in a database deadlock
since you are basically doing update promise.all

I think performance wise it would maby be better to use strapi.db.updateMany then check if you get back the right number of results if not then do a findMaby to check what one did not update

Reply:
There's no actual update being made here, it's just a frontend validation check before actually executing the action that doesn't exist yet.

Reply:
A fair sorry for making the wrong assumption here

Reply:
No worries thanks for swinging by and leaving a comment :)